src/librustc/hir/map/collector.rs

```diff
@@ -226,4 +226,8 @@ impl<'ast> Visitor<'ast> for NodeCollector<'ast> {
     fn visit_lifetime(&mut self, lifetime: &'ast Lifetime) {
         self.insert(lifetime.id, NodeLifetime(lifetime));
     }
+
+    fn visit_macro_def(&mut self, macro_def: &'ast MacroDef) {
+        self.insert_entry(macro_def.id, NotPresent);
```

Comment:
I think because the AST is no longer around when we generate metadata.

If we don't want to include exported macros in the HIR, I think the best alternative would be to move the AST's exported macros into a local  in `driver::compile_input` and pass them directly to the metadata encoder.